src/components/ShoppingList/ShoppingList.js

```diff
@@ -45,6 +46,27 @@ const ShoppingList = () => {
     setItems(updatedItems);
   };
 
+  const completeItem = (key) => {
+    const completedItems = items.map((item) => {
+      const newItem = item;
+      if (item.key === key) {
+        if (item.isCompleted === false) {
+          return {
+            ...item,
+            isCompleted: true,
+          };
+        }
+
+        return {
+          ...item,
+          isCompleted: false,
+        };
+      }
+      return newItem;
+    });
+    setItems(completedItems);
+  };
```

Comment:
Did you have a go at adding a test for items moving around? Let me know how you got on 😄

Reply:
I've done a test for the button press (the easy one I know!) and getting there with the remaining. I feel like I understand how to check if the "isCompleted" variable has toggled, but assessing the order of the items I'm not so sure.